app/models/storage.rb

```diff
@@ -70,6 +71,12 @@ class Storage < ApplicationRecord
 
   SUPPORTED_STORAGE_TYPES = %w( VMFS NFS FCP ISCSI GLUSTERFS )
 
+  supports :smartstate_analysis do
+    if ext_management_systems.blank? || !ext_management_system.kind_of?(ManageIQ::Providers::Vmware::InfraManager)
```

Comment:
Well the mount points are well defined for VMware so we can link up a path on the host to a datastore returned in inventory, the logic will be different for other providers